buildlib/pr/mad_tests.yml

```diff
@@ -0,0 +1,91 @@
+parameters:
+  HCA:
+
+jobs:
+  - job: Server_Up
+    pool:
+      name: MLNX
+      demands: mad_server
+    workspace:
+      clean: outputs
+    steps:
+      - checkout: self
+        clean: true
+        fetchDepth: 100
+        retryCountOnTaskFailure: 5
+      - bash: ./contrib/test_mad.sh run_mad_server
+        env:
+          HCA: ${{ parameters.HCA }}
+        displayName: Server Up          
+
+      - task: Bash@3
+        name: Set_Vars
+        inputs:
+          targetType: "inline"
+          script: ./contrib/test_mad.sh set_vars
```

Comment:
for my understanding: why not sourcing contrib/test_mad.sh then calling set_vars?

Reply:
Sourcing for each step is repetitive. Why not directly execute the required functions?

Reply:
I agree with Thomas, sourcing bash functions and then run them is best practice and what everyone is familier with, your implementation is a little bit confusing and very unusual.
However, i understand that if we need to source it each task/step it might look like repetitive work, but some time we can have repetitive in favor of readable code

Reply:
Done